### DISCUSSION TITLE
Possible contradiction in documentation for ProgramOptions

### DISCUSSION THREAD

**Discussion by christian-2** [cilium/ebpf#1329] [Q&A] [answered]
The documentation for [`ProgramOptions`](https://github.com/cilium/ebpf/blob/1a65b78df6d9fbe88cd546d689e3e1af54227f23/prog.go#L45) says regarding `LogLevel` "setting this to a non-zero value will _unconditionally_ enable the verifier log", and regarding `LogDisabled` "disables the verifier log completely, _regardless_ of other options".

Isn't there a contradition, i.e. is the verifier log enabled or disabled when `LogLevel` is non-zero _and_ `LogDisabled` is `true`?

**Comment by brycekahle** [comment, marked as answer]
`LogDisabled` takes precedence here

**Comment by christian-2** [comment]
Should this be clarified in the comments? If so, I could submit a (first) PR to that effect.

**Reply by lmb** [reply]
That would be nice!

**Comment by christian-2** [comment]
Here is another case: should the [sentence](https://pkg.go.dev/github.com/cilium/ebpf#MapOptions) "The base path to pin maps in if requested via `PinByName`" not refer to `PinPath` instead? Again, I can prepare a PR, if applicable.

**Reply by lmb** [reply]
A bit confusing maybe, this means that `PinPath` is only used if `MapSpec.Pinning` is `PinByName`.
